test/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDaoTest.java

```diff
@@ -690,9 +721,29 @@ public void getUploadSchemasForStudy() {
 
         // execute and validate
         List<UploadSchema> outputSchemaList = dao.getUploadSchemasForStudy(new StudyIdentifierImpl("test-study"));
-        assertSame(mockResult, outputSchemaList);
+        // No longer assertSame, as the DAO filters the list it gets from DDB mapper.
+        assertEquals(mockResult, outputSchemaList);
     }
+    
+    @Test
+    public void getUploadSchemasAllRevisions() {
+        // I want to test the actual sort order sent back by DDB, but all the tests here are mocks, so I'll
+        // continue down that path.
+        DynamoDBMapper mockMapper = setupMockMapperWithListResults();
+
+        // set up test dao
+        DynamoUploadSchemaDao dao = new DynamoUploadSchemaDao();
+        dao.setDdbMapper(mockMapper);
 
+        // execute and validate
+        List<UploadSchema> outputSchemaList = dao.getUploadSchemaAllRevisions(new StudyIdentifierImpl("test-study"), "AAA");
+
+        assertEquals(3, outputSchemaList.size());
+        assertEquals(1, outputSchemaList.get(0).getRevision());
+        assertEquals(2, outputSchemaList.get(1).getRevision());
+        assertEquals(3, outputSchemaList.get(2).getRevision());
```

Comment:
Note: In the actual query, you query for the schemas in reverse order, but in the mock, you return them in forward order.